### PR TITLE
Improve test args

### PR DESCRIPTION
**Pull request by tianleh**
### Description
Address db and Peter's remaining comments as follow up.
Related merged PR https://github.com/opensearch-project/opensearch-build/pull/1263 

### Issues Resolved
Resolves https://github.com/opensearch-project/opensearch-build/issues/1276
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tianleh** [comment]
For @dblock 's comment, 
https://github.com/opensearch-project/opensearch-build/pull/1263#discussion_r765833715

I followed it by using the `get` method with default value. We need to have `None` check against `args.paths` since it could be `None`.

In terms of moving validation logic to `parser`, based on my research, there are two ways to do it, either through `type` or `action`. (See this reference https://stackoverflow.com/questions/37471636/python-argument-parsing-validation-best-practices.)

I am very curious whether `type` can support a complicated structure like this.
```
type=Dict[str, TestArgsPathValidator.validate]
```

However, it didn't work. Thus I put the `TestArgsPathValidator.validate` inside the action `TestKwargs`

**Comment by tianleh** [comment]
For @peternied 's comment, I added the test manifest loading and read the content for setting Dashboards args. This indeed makes the unit test more clear since we just need to check `None` for Dashboards args in case of OpenSearch test manifest.

**Comment by tianleh** [comment]
During local test, `./test.sh` is not printing the logs in console. Need to check whether it is because of my local console configuration or some coding logic change.

File: src/test_workflow/test_args.py
Comment (dblock) [inline src/test_workflow/test_args.py:45]: Why do you need the `if`? Just assign it if it's passed in. Or better, assign `self.paths` and use `self.paths.get("...")` as needed.

File: tests/tests_test_workflow/data/test_manifest_opensearch_dashboards.yml
Comment (dblock) [inline tests/tests_test_workflow/data/test_manifest_opensearch_dashboards.yml:1]: Since it's a manifest I would use `-` instead of `_` in the filename.

**Review by dblock — APPROVED** [review]
Could you please fix the `if` before merge? Good work.

**Review by seraphjiang — APPROVED** [review]

**Review by peternied — CHANGES_REQUESTED** [review]
Do not merge until we have completed the production release testing, I'll unblock at that time

File: src/test_workflow/test_args.py
Comment (tianleh) [reply src/test_workflow/test_args.py:45]: E.g 
1) `./build.sh manifests/1.3.0/opensearch-1.3.0.yml`

2) `./build.sh manifests/1.3.0/opensearch-dashboards-1.3.0.yml`

Both will set the default value for OpenSearch. But only 2) will need to set a default value for Dashboards. I think this is also what Peter means in this comment https://github.com/opensearch-project/opensearch-build/pull/1263#discussion_r765899016

**Review by peternied — APPROVED** [review]
Production testing has completed unblocking this PR, thanks for your patience

**Comment by tianleh** [comment]
> Production testing has completed unblocking this PR, thanks for your patience

Thanks for your confirmation. I am still checking why the logging is not printed to console during execution. I will take care of this before merging this change.

File: src/test_workflow/test_args.py
Comment (dblock) [reply src/test_workflow/test_args.py:45]: First, default paths to `{}` and assign paths to `self`. This way it's always a dictionary and you don't have to do `if args.paths` ever.

```
self.paths = args.paths
```

Now, everywhere you need a dashboards path you could do `self.paths.get("opensearch-dashboards", cwd)`, or if you must, declare it as a property.

```
@property
def opensearch_dashboards_path(self)
    self.paths.get("opensearch-dashboards", cwd)
```

Does this remove the `if` or am I missing something?

**Comment by tianleh** [comment]
Somehow the logging config is changed by this PR. Console still cannot see the execution logs. No obvious clue which change is suspicious. I will comment out changes line by line to see which one causes the issue.

**Comment by tianleh** [comment]
> Somehow the logging config is changed by this PR. Console still cannot see the execution logs. No obvious clue which change is suspicious. I will comment out changes line by line to see which one causes the issue.

Finally finds the reason. 

This is where the log level is configured.

`    console.configure(level=args.logging_level)`

https://github.com/opensearch-project/opensearch-build/blob/main/src/run_integ_test.py#L19

However, this will not work if the preceding execution chains have ever called `logging.info` before it. In my previous commit, I added a logic to read manifest from path. https://github.com/opensearch-project/opensearch-build/blob/12093e78268c6052aeb798987ae97f0f770db1a7/src/manifests/manifest.py#L59 

This has a such preceding `logging.info` which thus failed the console output. This is no longer an issue after I applied db's suggestion. However, this is still an interesting knowledge to share.

File: src/test_workflow/test_args.py
Comment (tianleh) [reply src/test_workflow/test_args.py]: Great suggestion! Originally my mind was limited to solve this problem within the scope of `TestArgs` class. But your suggestion opened my mind to let the caller (who intrinsically knows the situation better than `TestArgs` ) decide the parameters to use. 

Just updated the code and it is much cleaner and elegant!

**Review by dblock — APPROVED** [review]

**Pull request merged**
